### ISSUE TITLE
Design Intent - Accessibility Feedback - Dependent Experience Team, Dependency Claims, VA Form 21-686C Military Base Outside the U.S. Address Issues

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/va.gov-team#68546]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Should:** When pre-filling content, ensure that any errors are communicated effectively to screen reader users.
  - You could show no errors on prefilled data, then only run the validation when the user interacts with the field or tries to submit. That would ensure the error appears and is announced to screen reader users, assuming the alert has the appropriate role.
  - If the error is shown on page load, the `aria-invalid` attribute should be set on the form. The error message should also be read out when a user tabs to the form element, which can be achieved through `aria-describedby`.
- **Consider:** Wherever possible, try to use a "one thing per page" approach rather than any sort of conditional display of information or form fields. There are a handful of situations where conditionally revealing new things makes sense, but usually breaking things into separate pages helps with cognitive load (keeps focus on a particular task) and avoids some tricky assistive technology interactions.
    - As discussed in the meeting, it follows the address pattern and is an existing form, so updating to a "one thing per page" pattern might not be feasible. But front-loading the military address checkbox and having two paths would make the flow clearer.
- **Consider:** The interaction of them checking the box for military base, then entering a US zip code, then they are instructed to go back and uncheck the box is a little confusing. I don't have a specific recommendation to fix that, but I'll put some options below that Matt and Erin mentioned.
    - One idea is checking the right boxes if you know the information was for a military address. This one is tricky because it may change information without them knowing.
    - Another was similar to entering a shipping address that gets validated on submit and shows a "you entered X, did you mean Y, would you like to use that" dialog.
